dot/parachain/prospective-parachains/fragment_chain.go

```diff
@@ -591,16 +630,16 @@ func (f *fragmentChain) candidateBacked(newlyBackedCandidate parachaintypes.Cand
 }
 
 // canAddCandidateAsPotential checks if this candidate could be added in the future
-func (f *fragmentChain) canAddCandidateAsPotential(entry *candidateEntry) error {
-	candidateHash := entry.candidateHash
+func (f *fragmentChain) canAddCandidateAsPotential(candidate HypoteticalOrConcrete) error {
```

Comment:
can you provide more details about this argument type? what are all possible types we can pass here?
As of now i see only 1 `candidateEntry`.

Reply:
https://github.com/ChainSafe/gossamer/pull/4641#discussion_r2045183696